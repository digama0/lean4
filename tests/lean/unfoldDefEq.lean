def Wrapper (n : Nat) : Type Ã— Type :=
  (Fin n, Fin n)

def SomeProperty {n} (x : Fin n) : Prop :=
  True

example (x : (Wrapper n).1) (h : SomeProperty x) : True := by
  unfold Wrapper at x
  trace_state
  simp at x
  trace_state
  sorry

example (x : (Wrapper n).1) (h : SomeProperty x) : True := by
  simp [Wrapper] at x
  trace_state
  sorry
