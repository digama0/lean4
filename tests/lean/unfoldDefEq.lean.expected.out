unfoldDefEq.lean:7:0-7:7: warning: declaration uses 'sorry'
n : Nat
x : (Fin n, Fin n).fst
h : SomeProperty x
⊢ True
n : Nat
x : Fin n
h : SomeProperty x
⊢ True
unfoldDefEq.lean:14:0-14:7: warning: declaration uses 'sorry'
n : Nat
x : Fin n
h : SomeProperty x
⊢ True
